src/api/WeatherComponentAPI.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import axios, { AxiosResponse } from 'axios';
+import { weatherAPIKey } from '../key.json';
+
+const WeatherComponentAPI: React.FC<unknown> = (city) => {
+  //Property 'city' does not exist on type '{ children?: ReactNode; }'
+  const [weather, setWeather] = React.useState<string>('');
+  const [weatherIcon, setWeatherIcon] = React.useState<string>('');
+  const [temperature, setTemperature] = React.useState<string>('');
+
+  React.useEffect(() => {
+    axios
+      .get<any, AxiosResponse<any, any>, any>(
+        `http://api.weatherapi.com/v1/current.json?key=${weatherAPIKey}&q=${city}`
```

Comment:
Think you can put this part into a separate file in /api

Reply:
oh it is here, I think we should leave only function that fetches data here and all the react code to components/